osquery/tables/networking/linux/interface_bonds.cpp

```diff
@@ -0,0 +1,153 @@
+/*
+ *  Copyright (c) 2014, Facebook, Inc.
+ *  All rights reserved.
+ *
+ *  This source code is licensed under the BSD-style license found in the
+ *  LICENSE file in the root directory of this source tree. An additional grant
+ *  of patent rights can be found in the PATENTS file in the same directory.
+ *
+ */
+
+#include <string>
+
+#include <boost/algorithm/string/join.hpp>
+#include <boost/algorithm/string/predicate.hpp>
+
+#include <osquery/tables.h>
+
+#include "osquery/tables/networking/utils.h"
+
+namespace osquery {
+namespace tables {
+
+QueryData genInterfaceBonds(QueryContext& context) {
+  Row r;
+  QueryData results;
+
+/* Read data from /sys/class/net/bonding_masters file and check for the bond name.*/
+FILE *read,*read2,*read3,*read4,*read5,*read6,*read7,*read8,*read9,*read10,*read11,*read12,*read13,*read14,*read15;
+
+      read = fopen("/sys/class/net/bonding_masters","r");
```

Comment:
Prefer to use the osquery `readFile` APIs. Check out https://github.com/facebook/osquery/blob/master/include/osquery/filesystem.h#L71

What you'll end up with is something similar to:
```cpp
std::string content;
auto status = readFile("/sys/class/net/bonding_masters", content);
r["bond"] = content;
```